common/services.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package common
```

Comment:
I think we need to re-think how we are using `common` package, there are a growing number of unrelated files in this package which makes it difficult to understand its purpose. Great article that recommends avoiding this pitfall, see [here](https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common).

Reply:
yeah I think this service stuff could be in its own package? I figured common was more for types and useful functions

Reply:
perhaps I should move it into `dot` or `internal`?

Reply:
Moved into internal/services; created internal/api